server/src/utils/misc.ts

```diff
@@ -63,7 +63,7 @@ export const isSmartSearchEnabled = (machineLearning: SystemConfig['machineLearn
 export const isFacialRecognitionEnabled = (machineLearning: SystemConfig['machineLearning']) =>
   isMachineLearningEnabled(machineLearning) && machineLearning.facialRecognition.enabled;
 export const isDuplicateDetectionEnabled = (machineLearning: SystemConfig['machineLearning']) =>
-  isMachineLearningEnabled(machineLearning) && machineLearning.duplicateDetection.enabled;
+  isMachineLearningEnabled(machineLearning) && isSmartSearchEnabled(machineLearning) && machineLearning.duplicateDetection.enabled;
```

Comment:
```suggestion
  isMachineLearningEnabled(machineLearning) && machineLearning.clip.enabled && machineLearning.duplicateDetection.enabled;
```
This should also accomplish the goal, someone who know about making good code should determine whether my change or this alternative change is better but they should both be identical on functionality.

Reply:
@mertalev, you might know which is best since you made the Duplicate Detection and know a lot about the AI features of immich.

Reply:
I think you can just change `isMachineLearningEnabled` in the original check to `isSmartSearchEnabled`.

Reply:
I'll test that and get back to you.

Reply:
> I think you can just change `isMachineLearningEnabled` in the original check to `isSmartSearchEnabled`.

This works.
```suggestion
  isSmartSearchEnabled(machineLearning) && machineLearning.duplicateDetection.enabled;
```